utils/parser.go

```diff
@@ -0,0 +1,46 @@
+package utils
+
+import (
+	"encoding/json"
+	"errors"
+	"log"
+)
+
+//reference human readable keys to DB key values
+var keyReferences = map[string]string{"id": "id", "createdDate": "created_at", "name": "name"}
+
+//ParseFilters is used to parse filters form JSON string to a map of (string,string)
+//If any required key is missing, it is set as default
+//All unknown keys are ignored
+func ParseFilters(filterString string) (map[string]string, error) {
+	if filterString == "" {
+		filterString = "{}"
+	}
+	filters := map[string]string{}
+	err := json.Unmarshal([]byte(filterString), &filters)
+	var cleanedFilters map[string]string
+	if err == nil {
+		cleanedFilters, err = cleanFilters(filters)
+	}
+	if err != nil {
+		log.Println(err)
+		return make(map[string]string), errors.New("Illegal filter syntax")
+	}
+	return cleanedFilters, nil
+}
+
+func cleanFilters(filters map[string]string) (map[string]string, error) {
+	supportedKeys := []string{"id", "createdDate", "name"}
+	cleanedFilters := make(map[string]string)
+	for _, key := range supportedKeys {
+		value := filters[key]
+		if value != "asc" && value != "desc" && value != "" {
```

Comment:
Should it be case insensitive?

Reply:
What happens down the method if the string is empty? Are you setting a default? It doesn't look like

Reply:
yes it can be case insensitive ( do we want it to be ?)

Go by default assigns "" to keys that are missing so just to verify validity we need to check empty too

Reply:
but for keys that are valid and value is empty what happens? Also, yes if DB support case insensitivity, which I think it does, it should be. Or we should force lowercase conversion before parsing the string.